core/services/config.go

```diff
@@ -0,0 +1,250 @@
+package services
+
+import (
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+
+	"github.com/go-skynet/LocalAI/core/schema"
+	"github.com/go-skynet/LocalAI/pkg/downloader"
+	"github.com/go-skynet/LocalAI/pkg/utils"
+	"github.com/rs/zerolog/log"
+	"gopkg.in/yaml.v3"
+)
+
+type ConfigLoader struct {
+	configs map[string]schema.Config
+	sync.Mutex
+}
+
+// Load a config file for a model
+func LoadConfigFileByName(modelName, modelPath string, cl *ConfigLoader, debug bool, threads, ctx int, f16 bool) (*schema.Config, error) {
```

Comment:
ditto - the config loader probably should just sit next to the application configuration